android-sdk/src/main/java/com/mobilecoin/lib/DestinationMemo.java

```diff
@@ -9,9 +9,9 @@
 import java.util.Objects;
 
 /** Represents a destination memo, which corresponds to the "DestinationMemo" specification. */
-public final class DestinationMemo extends TxOutMemo {
```

Comment:
Why not keep it final?

Reply:
Initially, DestinationWithPaymentRequestMemo and DestinationWithPaymentIntentMemo were going to extend DestinationMemo. The goal was to let clients only handle generic Destion and Sender memos.

The issue with this is that it doesn't work well with how the original RTH spec was finalized. We can't just get the memo data from the memo class. Each memo type has it's own memo data class and an appropriately named method to access it. So if we just make, for example DestinationWithPaymentIntentMemoData extend DestinationMemoData, then DestinationWithPaymentIntentMemo would need to provide two accessors: getDestinatonWithPaymentIntentMemoData() and getDestinationMemoData() which would both return the same object. This is unnecessarily confusing. I think this behavior would best be implemented with a new high-level API spec.

I can change this and all the other memo types to final, if we don't need them to be extended. But I don't know if there isna reason they need to be.

Reply:
Not having `final` signals to the SDK users that the author intends for them to extend this class to work with it. If that's not your intention, then make it `final`. Also, simple overrides can have serious side effects, and non-final classes must take that into account, so unless you expect users to extend it, the safer way is to declare it `final`. (e.g. a private method calls a public method in the base class, then overrides the public method, etc)